server/src/test/java/io/spine/server/bc/BoundedContextShould.java

```diff
@@ -337,4 +351,53 @@ public void do_not_expose_invisible_aggregate() {
         assertFalse(boundedContext.findRepository(SecretProject.class)
                                   .isPresent());
     }
+
+    @Test(expected = NullPointerException.class)
+    public void throw_NPE_when_default_state_is_null() {
+        final ProjectAggregateRepository repository = new ProjectAggregateRepository();
+        final Map mockMap = mock(Map.class);
+        when(mockMap.get(any())).thenReturn(null);
+        final Object defaultStateRegistry = getObjectFromNestedEnumField(
+                DEFAULT_STATE_REGISTRY_FULL_CLASS_NAME, DEFAULT_STATE_REGISTRY_SINGLETON_FIELD_NAME);
+        injectField(defaultStateRegistry, DEFAULT_STATES_FIELD_NAME, mockMap);
+
+        try {
+            boundedContext.register(repository);
+        }catch (NullPointerException e) {
+            // reassigning mock map to real map, to prevent failing other tests
+            final Map<Class<? extends Entity>, Message> defaultState = newConcurrentMap();
+            injectField(defaultStateRegistry, DEFAULT_STATES_FIELD_NAME, defaultState);
+            throw e;
+        }
+    }
+
+    private Object getObjectFromNestedEnumField(String fullClassName, String fieldName) {
+        Object result = null;
+        try {
+            final Class<?> aClass = Class.forName(fullClassName);
+            final Object enumConstant = aClass.getEnumConstants()[0];
+            final Field field = aClass.getDeclaredField(fieldName);
+            field.setAccessible(true);
+            result = field.get(enumConstant);
+        } catch (ClassNotFoundException e) {
+            fail("Class " + fullClassName + " not found.");
+        } catch (NoSuchFieldException e) {
+            fail("Field " + fieldName + " should exist.");
+        } catch (IllegalAccessException e) {
+            fail(e.getMessage());
+        }
+        assertNotNull(result);
+        return result;
+    }
+
+    private void injectField(Object target, String fieldName, Object injectableValue) {
```

Comment:
`injectableValue` --> `valueToInject`. It feels a bit different and will suit better in this case.